src/UIKit/UIGestureRecognizer.cs

```diff
@@ -130,12 +132,22 @@ public void RemoveTarget (Token token)
 				throw new ArgumentNullException ("token");
 			if (recognizers == null)
 				return;
-			if (recognizers == token)
-				recognizers = null;
-			Hashtable asHash = recognizers as Hashtable;
-			if (asHash != null)
-				asHash.Remove (token);
-			RemoveTarget (token, token is ParametrizedDispatch ? Selector.GetHandle (parametrized_selector) : Selector.GetHandle (tsel));
+			if (recognizers.ContainsKey (token)){
+				var sel = recognizers [token];
+				recognizers.Remove (token);
+				RemoveTarget (token, sel);
+			}
+		}
+
+		//
+		// Used to enumerate all the registered handlers for this UIGestureRecognizer
+		//
+		public IEnumerable<Token> GetTargets ()
+		{
+			if (recognizers == null)
```

Comment:
Can't this just be:

```csharp
public IEnumerable<Token> GetTargets ()
{
    return recognizers?.Keys;
}
```

Reply:
Good idea to avoid the for loop, but I will keep the split yield break/yield return to avoid returning a null for empty cases, instead, by catching this early we would return an empty collection.

Reply:
`recognizers?.Keys ?? Array.Empty<Token>`? That would avoid allocating an Iterator for the method.